datalad/metadata/search.py

```diff
@@ -181,15 +181,14 @@ def _get_search_schema(ds):
                 # might be different when some aggregated metadata was
                 # generated with an old version of datalad
                 # in this case we should actually load the old vocabulary
-                set.add(', '.join(i for i in v) if isinstance(v, (tuple, list)) else v)
+                #set.add(', '.join(i for i in v) if isinstance(v, (tuple, list)) else v)
```

Comment:
Not sure what state of my brain can be held responsible here, but I think this should be `v = (.. if ... else v)` to avoid non-string/non-dict values in key definitions later on.